server/src/internalClusterTest/java/org/opensearch/gateway/remote/RemoteRoutingTableServiceIT.java

```diff
@@ -0,0 +1,288 @@
+        BlobStoreRepository repository = prepareClusterAndVerifyRepository();
+
+        RemoteClusterStateService remoteClusterStateService = internalCluster().getClusterManagerNodeInstance(
+            RemoteClusterStateService.class
+        );
+        RemoteManifestManager remoteManifestManager = remoteClusterStateService.getRemoteManifestManager();
+        verifyUpdatesInManifestFile(remoteManifestManager);
+
+        List<RoutingTable> routingTableVersions = getRoutingTableFromAllNodes();
+        assertTrue(areRoutingTablesSame(routingTableVersions));
+
+        // Update index settings
+        updateIndexSettings(INDEX_NAME, IndexMetadata.SETTING_NUMBER_OF_REPLICAS, 2);
+        ensureGreen(INDEX_NAME);
+        assertBusy(() -> {
+            int indexRoutingFilesAfterUpdate = repository.blobStore().blobContainer(indexRoutingPath).listBlobs().size();
+            // At-least 3 new index routing files will be created as shards will transition from INIT -> UNASSIGNED -> STARTED state
+            assertTrue(indexRoutingFilesAfterUpdate >= indexRoutingFiles.get() + 3);
+        });
+
+        verifyUpdatesInManifestFile(remoteManifestManager);
+
+        routingTableVersions = getRoutingTableFromAllNodes();
+        assertTrue(areRoutingTablesSame(routingTableVersions));
+
+        String clusterManagerNode = internalCluster().getClusterManagerName();
+        NodesStatsResponse nodesStatsResponse = client().admin()
+            .cluster()
+            .prepareNodesStats(clusterManagerNode)
+            .addMetric(NodesStatsRequest.Metric.DISCOVERY.metricName())
+            .get();
+        // Delete the index and assert its deletion
+        deleteIndexAndVerify(remoteManifestManager);
+
+        routingTableVersions = getRoutingTableFromAllNodes();
+        assertTrue(areRoutingTablesSame(routingTableVersions));
+    }
+
+    public void testRemoteRoutingTableIndexNodeRestart() throws Exception {
+        BlobStoreRepository repository = prepareClusterAndVerifyRepository();
+
+        List<RoutingTable> routingTableVersions = getRoutingTableFromAllNodes();
+        assertTrue(areRoutingTablesSame(routingTableVersions));
+
+        // Ensure node comes healthy after restart
+        Set<String> dataNodes = internalCluster().getDataNodeNames();
+        internalCluster().restartNode(randomFrom(dataNodes));
+        ensureGreen();
+        ensureGreen(INDEX_NAME);
+
+        // ensure restarted node joins and the cluster is stable
+        assertEquals(3, internalCluster().clusterService().state().nodes().getDataNodes().size());
+        ensureStableCluster(4);
+        assertRemoteStoreRepositoryOnAllNodes(REMOTE_ROUTING_TABLE_REPO);
+
+        assertBusy(() -> {
+            int indexRoutingFilesAfterNodeDrop = repository.blobStore().blobContainer(indexRoutingPath).listBlobs().size();
+            assertTrue(indexRoutingFilesAfterNodeDrop > indexRoutingFiles.get());
+        });
+
```

Comment:
Did we run iterations for this test as well? Lets ensure this is also always true.

Reply:
Yes, ran 100 iterations over all tests.